service_config.go

```diff
@@ -400,3 +401,34 @@ func getMaxSize(mcMax, doptMax *int, defaultVal int) *int {
 func newInt(b int) *int {
 	return &b
 }
+
+func init() {
+	internal.EqualServiceConfigForTesting = equalServiceConfig
+}
+
+// equalServiceConfig compares two configs. The rawJSONString field is ignored,
+// because they may diff in white spaces.
+//
+// If any of them is NOT *ServiceConfig, return false.
+func equalServiceConfig(a, b serviceconfig.Config) bool {
```

Comment:
Is it really required that this function be exposed though the internal package?
I see that this only used by a couple of xds_tests. Could we rather move this to xds/testutils?

Reply:
To clear the `rawJSONString` field, it has to be here.
Or, we export a method to clear the field...